plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/SortFields.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.plugin.iceberg.sort.CaseInsensitiveStream;
+import io.trino.plugin.iceberg.sort.SortDefinitionBaseVisitor;
+import io.trino.plugin.iceberg.sort.SortDefinitionLexer;
+import io.trino.plugin.iceberg.sort.SortDefinitionParser;
+import io.trino.spi.TrinoException;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.atn.PredictionMode;
+import org.apache.iceberg.NullOrder;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortField;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.SortOrderBuilder;
+
+import java.util.List;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.iceberg.IcebergTableProperties.SORTED_BY_PROPERTY;
+import static io.trino.plugin.iceberg.PartitionFields.fromIdentifierToColumn;
+import static io.trino.plugin.iceberg.PartitionFields.quotedName;
+import static io.trino.spi.StandardErrorCode.INVALID_TABLE_PROPERTY;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class SortFields
+{
+    private static final BaseErrorListener ERROR_LISTENER = new BaseErrorListener()
+    {
+        @Override
+        public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String message, RecognitionException e)
+        {
+            throw new TrinoException(INVALID_TABLE_PROPERTY, "Unable to parse sort field", e);
+        }
+    };
+
+    private SortFields() {}
+
+    public static SortOrder parseSortFields(Schema schema, List<String> fields)
+    {
+        SortOrder.Builder builder = SortOrder.builderFor(schema);
+        parseSortFields(builder, fields);
+        try {
+            return builder.build();
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(INVALID_TABLE_PROPERTY, "Invalid " + SORTED_BY_PROPERTY + " definition", e);
+        }
+    }
+
+    public static void parseSortFields(SortOrderBuilder<?> sortOrderBuilder, List<String> fields)
```

Comment:
TBH I think that this method is not needed. It doesn't do much and is only used in a method above(if this is not correct then please ignore this comment). Also because it is overloaded it makes the picture less clear at least for me

Reply:
It's used one other place, in the `IcebergMetadata` code that changes the sort order when you use `SET PROPERTIES`